lib/marin/src/marin/rl/health_monitor.py

```diff
@@ -0,0 +1,591 @@
+# Copyright 2025 The Marin Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Health monitoring system for RL curriculum learning.
+
+This module provides comprehensive monitoring of environment performance,
+automated warning generation, and regression detection for graduated environments.
+"""
+
+import logging
+import time
+from collections import defaultdict, deque
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import Any
+
+import numpy as np
+
+logger = logging.getLogger(__name__)
+
+
+class WarningType(Enum):
+    """Types of warnings that can be generated."""
+
+    GRADUATED_REGRESSION = "graduated_regression"
+    TRAINING_STAGNATION = "training_stagnation"
+    PERFORMANCE_VOLATILITY = "performance_volatility"
+    UNEXPECTED_TRANSITION = "unexpected_transition"
+    PROLONGED_TRAINING = "prolonged_training"
+
+
+class HealthStatus(Enum):
+    """Health status levels for environments."""
+
+    HEALTHY = "healthy"
+    WARNING = "warning"
+    CRITICAL = "critical"
+
+
+@dataclass
+class HealthWarning:
+    """A warning about environment health."""
+
+    type: WarningType
+    env_id: str
+    message: str
+    severity: str  # "low", "medium", "high", "critical"
```

Comment:
why a string here but the HealthStatus enum above? use that? WarningSeverity enum?